sktime/forecasting/reconcile.py

```diff
@@ -0,0 +1,355 @@
+# -*- coding: utf-8 -*-
+# !/usr/bin/env python3 -u
+# copyright: sktime developers, BSD-3-Clause License (see LICENSE file)
+"""Implements reconciled forecasters for hierarchical data."""
+
+__all__ = ["ReconcilerForecaster"]
+__author__ = [
+    "ciaran-g",
+]
+
+# todo: include the reconciler transformers?
+# todo: top down historical proportions?
+
+from warnings import warn
+
+import numpy as np
+import pandas as pd
+from numpy.linalg import inv
+
+from sktime.forecasting.base import BaseForecaster, ForecastingHorizon
+from sktime.transformations.hierarchical.aggregate import _check_index_no_total
+from sktime.transformations.hierarchical.reconcile import (
+    _get_s_matrix,
+    _parent_child_df,
+)
+
+
+class ReconcilerForecaster(BaseForecaster):
+    """Hierarchical reconcilation forecaster.
+
+    Reconciliation is applied to make the forecasts in a hierarchy of
+    time-series sum together appropriately.
+
```

Comment:
maybe a few more lines how it relates to the `forecaster` arg, i.e., that forecasts are made by `forecaster` and then etc

Reply:
added a few more lines